dev-libs/pigpio/files/pigpio-66-makefile.patch

```diff
@@ -0,0 +1,117 @@
++LIB3     = libpigpiod_if2.so.0
+ OBJ3     = pigpiod_if2.o command.o
+ 
+ LIB      = $(LIB1) $(LIB2) $(LIB3)
+ 
+-ALL     = $(LIB) x_pigpio x_pigpiod_if x_pigpiod_if2 pig2vcd pigpiod pigs
++ALL     = $(LIB) pig2vcd pigpiod pigs
+ 
+-LL1      = -L. -lpigpio -pthread -lrt
++LL1      = -L. $(LIB1) -pthread -lrt
+ 
+-LL2      = -L. -lpigpiod_if -pthread -lrt
++LL2      = -L. $(LIB2) -pthread -lrt
+ 
+-LL3      = -L. -lpigpiod_if2 -pthread -lrt
++LL3      = -L. $(LIB3) -pthread -lrt
+ 
+-prefix = /usr/local
++prefix = /usr
+ exec_prefix = $(prefix)
+ bindir = $(exec_prefix)/bin
+ includedir = $(prefix)/include
+ libdir = $(prefix)/lib
+-mandir = $(prefix)/man
++mandir = $(prefix)/share/man
+ 
+ all:	$(ALL)
+ 
+@@ -64,41 +64,37 @@
+ 	$(CC) -o x_pigpiod_if2 x_pigpiod_if2.o $(LL3)
+ 
+ pigpiod:	pigpiod.o $(LIB1)
+-	$(CC) -o pigpiod pigpiod.o $(LL1)
+-	$(STRIP) pigpiod
++	$(CC) $(CFLAGS) $(LDFLAGS) -o pigpiod pigpiod.o $(LL1)
+ 
+ pigs:		pigs.o command.o
+-	$(CC) -o pigs pigs.o command.o
+-	$(STRIP) pigs
++	$(CC) $(CFLAGS) $(LDFLAGS) -o pigs pigs.o command.o
+ 
+ pig2vcd:	pig2vcd.o
+-	$(CC) -o pig2vcd pig2vcd.o
+-	$(STRIP) pig2vcd
++	$(CC) $(CFLAGS) $(LDFLAGS) -o pig2vcd pig2vcd.o
+ 
+ clean:
+ 	rm -f *.o *.i *.s *~ $(ALL)
+ 
+ install:	$(ALL)
+-	install -m 0755 -d                $(DESTDIR)/opt/pigpio/cgi
+ 	install -m 0755 -d                $(DESTDIR)$(includedir)
+ 	install -m 0644 pigpio.h          $(DESTDIR)$(includedir)
+ 	install -m 0644 pigpiod_if.h      $(DESTDIR)$(includedir)
+ 	install -m 0644 pigpiod_if2.h     $(DESTDIR)$(includedir)
+ 	install -m 0755 -d                $(DESTDIR)$(libdir)
+-	install -m 0755 libpigpio.so      $(DESTDIR)$(libdir)
+-	install -m 0755 libpigpiod_if.so  $(DESTDIR)$(libdir)
+-	install -m 0755 libpigpiod_if2.so $(DESTDIR)$(libdir)
++	install -m 0755 libpigpio.so.0      $(DESTDIR)$(libdir)
++	install -m 0755 libpigpiod_if.so.0  $(DESTDIR)$(libdir)
++	install -m 0755 libpigpiod_if2.so.0 $(DESTDIR)$(libdir)
++	ln -srf $(DESTDIR)$(libdir)/libpigpio.so.0 $(DESTDIR)$(libdir)/libpigpio.so
++	ln -srf $(DESTDIR)$(libdir)/libpigpiod_if.so.0 $(DESTDIR)$(libdir)/libpigpiod_if.so
++	ln -srf $(DESTDIR)$(libdir)/libpigpiod_if2.so.0 $(DESTDIR)$(libdir)/libpigpiod_if2.so
+ 	install -m 0755 -d                $(DESTDIR)$(bindir)
+ 	install -m 0755 pig2vcd           $(DESTDIR)$(bindir)
+ 	install -m 0755 pigpiod           $(DESTDIR)$(bindir)
+ 	install -m 0755 pigs              $(DESTDIR)$(bindir)
+-	if which python2; then python2 setup.py install; fi
+-	if which python3; then python3 setup.py install; fi
+ 	install -m 0755 -d                $(DESTDIR)$(mandir)/man1
+ 	install -m 0644 *.1               $(DESTDIR)$(mandir)/man1
+ 	install -m 0755 -d                $(DESTDIR)$(mandir)/man3
+ 	install -m 0644 *.3               $(DESTDIR)$(mandir)/man3
+-	ldconfig
+ 
+ uninstall:
+ 	rm -f $(DESTDIR)$(includedir)/pigpio.h
+@@ -117,19 +113,13 @@
+ 	ldconfig
+ 
+ $(LIB1):	$(OBJ1)
+-	$(SHLIB) -o $(LIB1) $(OBJ1)
+-	$(STRIPLIB) $(LIB1)
```

Comment:
I suppose you won't be able to upstream removals of stripping, so easy hack:

    emake ... STRIP=: STRIPLIB=: SIZE=:

I suppose you could also make python & ldconfig disable-able by adding extra vars for those tools.

Reply:
So if i understand you correctly you mean replacing/patching the occurrences of python and ldconfig in the Makefile using a variable that I set (or better disable) via emake similar to the STRIP and SIZE?
That would initially make the patch bigger than just deleting the lines, right?
Perhaps if i get that pushed upstream it would improve things.

Reply:
Yes, I'm going forward and talking about something you could submit upstream to avoid having to patch long-term.